### PR TITLE
[amp-story] Add support for different kinds of action icons inside the tooltip

### PR DESCRIPTION
**Pull request by Enriqe**
Closes: #20001

Part of: #19213 #16522

Extends support for different kind of icons on the tooltip.  
The icon on the left of the tooltip will be called: `componentIcon`, and on the right `actionIcon`.

(Follow ups will be send for i18n of the default tooltip text).

e.g.

image

image

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-embedded-component.js [hunk 1]
```diff
@@ -25,38 +25,103 @@ import {EventType, dispatch} from './events';
...
+};
+
+/**
+ * Components that can be expanded.
+ * @const {!Object}
+ * @private
  */
 const EXPANDABLE_COMPONENTS = {
-  TWITTER: 'amp-twitter',
+  TWITTER: {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-embedded-component.js]: As seen offline, could you please look into if we can reuse the ones we already have for the share icons? That'd save some bytes!

File: extensions/amp-story/1.0/amp-story-embedded-component.js [hunk 2]
```diff
@@ -381,7 +450,7 @@ export class AmpStoryEmbeddedComponent {
       return;
     }
 
-    if (Object.values(EXPANDABLE_COMPONENTS)
+    if (Object.values(getComponentSelectors(EXPANDABLE_COMPONENTS))
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-embedded-component.js]: See other comment

File: extensions/amp-story/1.0/amp-story-embedded-component.js [hunk 3]
```diff
@@ -418,6 +487,21 @@ export class AmpStoryEmbeddedComponent {
...
+   */
+  getEmbedConfigFor_(target) {
+    const componentIndex = findIndex(Object.values(INTERACTIVE_COMPONENTS),
+        config => {
+          return matches(target, config.selector);
+        });
+
+    return INTERACTIVE_COMPONENTS[
+        Object.keys(INTERACTIVE_COMPONENTS)[componentIndex]];
+  }
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-embedded-component.js]: At this point, it becomes pretty hard to know how the config is used (Object.values / Object.keys / ...). Object.keys and Object.values loop on the array, findIndex too, but it's mostly becoming hard to follow.
You have much more context on all this, but do you think we could avoid all these by using the `target.tagName` as keys of the config?
I was hoping that this could remove the need for the `getComponentSelectors` function, and would make this method as easy as something like `return COMPONENTS[target.tagName.toLowerCase()];`

File: extensions/amp-story/1.0/amp-story-embedded-component.js [hunk 4]
```diff
@@ -470,19 +556,38 @@ export class AmpStoryEmbeddedComponent {
...
+   * @private
+   */
+  updateTooltipActionIcon_(embedConfig) {
+    const actionIcon =
+      this.tooltip_.querySelector('.i-amphtml-tooltip-action-icon');
+
+    this.resources_.mutateElement(devAssert(actionIcon), () => {
+      Object.values(ActionIcon).map(iconClass => {
+        actionIcon.classList.toggle(iconClass, false);
+      });
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-embedded-component.js]: This loops twice on the array and does as many operations on classList that we have icons. Can we find a better solution, like maybe removing the class from `clearTooltip` earlier?

File: extensions/amp-story/1.0/amp-story-embedded-component.js (hunk 3 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/amp-story-embedded-component.js]: As chatted offline: We want to support both selectors and tag names for our config, which introduces some overhead at lookup. 

But I agree we are able to cut some complexity by using tag names in the keys and still check for their selectors once we get their config. 
Re. `getComponentSelectors` we unfortunately still need it to export the list of selectors to `page-advancement` so that it knows that it should delegate to this class. (On the upside, it only runs once).

Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-embedded-component.js]: Wow that's so much easier, thanks for improving this!

File: extensions/amp-story/1.0/amp-story-embedded-component.js [hunk 5]
```diff
@@ -25,38 +25,102 @@ import {EventType, dispatch} from './events';
...
+    'nodd"><path d="M0 0h400v400H0z"/><path fill="%231da1f2" fill-rule="nonz' +
+    'ero" d="M153.62 301.59c94.34 0 145.94-78.16 145.94-145.94 0-2.22 0-4.43' +
+    '-.15-6.63A104.36 104.36 0 0 0 325 122.47a102.38 102.38 0 0 1-29.46 8.07 ' +
+    '51.47 51.47 0 0 0 22.55-28.37 102.79 102.79 0 0 1-32.57 12.45c-15.9-16.' +
+    '906-41.163-21.044-61.625-10.093-20.461 10.95-31.032 34.266-25.785 56.87' +
+    '3A145.62 145.62 0 0 1 92.4 107.81c-13.614 23.436-6.66 53.419 15.88 68.4' +
+    '7A50.91 50.91 0 0 1 85 169.86v.65c.007 24.416 17.218 45.445 41.15 50.28' +
+    'a51.21 51.21 0 0 1-23.16.88c6.72 20.894 25.976 35.208 47.92 35.62a102.9' +
+    '2 102.92 0 0 1-63.7 22 104.41 104.41 0 0 1-12.21-.74 145.21 145.21 0 0 ' +
+    '0 78.62 23"/></g></svg>',
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-embedded-component.js]: Nit: Can we keep these in the CSS with the other launch icons? This will get big, especially as we start allowing many many components
